src/backend/distributed/utils/connection_cache.c

```diff
@@ -60,78 +51,27 @@ static void ReportRemoteError(PGconn *connection, PGresult *result, bool raiseEr
 PGconn *
 GetOrEstablishConnection(char *nodeName, int32 nodePort)
 {
+	int connectionFlags = SESSION_LIFESPAN;
```

Comment:
I'm ok with moving ahead, but noting that I still find the use of this flag pretty weird and it will likely cause bugs further down the line. The API currently doesn't guarantee that there is only a single sessionLifespan connection. For example, you could open a regular connection, claim it exclusively, open a new `SESSION_LIFESPAN` connection, unclaim the first connection, ask for a `SESSION_LIFESPAN` connection, and the API returns the first, now unclaimed connection with sessionLifespan marked as true. Now there are 2 session lifespan connections, which probably wasn't intended.